pandas/tests/test_generic.py

```diff
@@ -1539,6 +1540,15 @@ def test_to_xarray(self):
                            expected,
                            check_index_type=False)
 
+    def test_deepcopy_empty(self):
+        # This test covers empty frame copying with non-empty column sets
+        # as reported in issue #15370
+        # https://github.com/pandas-dev/pandas/issues/15370
+        empty_frame = DataFrame(data=[], index=[], columns=['A'])
+        empty_frame_copy = deepcopy(empty_frame)
+
+        self.assertEqual(empty_frame, empty_frame_copy)
```

Comment:
Just to be clear, when you say `self.deepcopy` do you mean `self.__deepcopy__`?  There is not currently an exposed function of that name.